### DISCUSSION TITLE
How to Modify the Search Bar (ui.input) Behavior in Table

### DISCUSSION THREAD

**Discussion by happybeginning1** [zauberzeug/nicegui#3833] [Q&A] [answered]
### Question

Hi,

The following NiceGUI code works well for filtering the table. I have a few questions about refining the search functionality.

```python
from nicegui import ui

columns = [
    {'name': 'name', 'label': 'Name', 'field': 'name', 'required': True, 'align': 'left'},
    {'name': 'age', 'label': 'Age', 'field': 'age', 'sortable': True},
]
rows = [
    {'name': 'Alice', 'age': 18},
    {'name': 'Bob', 'age': 21},
    {'name': 'Carol'},
]

table = ui.table(columns=columns, rows=rows, row_key='name')
ui.input().bind_value_to(table, 'filter')

ui.run()
```

1. The current search functionality works well overall, but I want the table to be filtered only after the user presses the Enter key in the input field. How can I achieve this?
2. When the user enters a keyword (e.g., "bob") and presses Enter to search, the table updates as expected. If the user then clears the input field, I would like the table to show all the rows automatically without requiring another Enter press. How can I implement this?
3. Is it possible to restrict the search to a specific column (e.g., only searching within the "name" column)? If so, how can I achieve this?

Thank you very much for your help!

**Comment by python-and-novella** [comment, marked as answer]
For 1 and 2, you can hook keypress.enter and value_change events:
```python3
from nicegui import ui

columns = [
    {'name': 'name', 'label': 'Name', 'field': 'name', 'required': True, 'align': 'left'},
    {'name': 'age', 'label': 'Age', 'field': 'age', 'sortable': True},
]
rows = [
    {'name': 'Alice', 'age': 18},
    {'name': 'Bob', 'age': 21},
    {'name': 'Carol'},
]

table = ui.table(columns=columns, rows=rows, row_key='name')

inp = ui.input()
inp.on('keypress.enter',lambda e:table.set_filter(e.sender.value))
inp.on_value_change(lambda e: table.set_filter(None) if len(e.sender.value)==0 else None)

ui.run()
```

**Reply by happybeginning1** [reply]
Thanks @python-and-fiction,
Your solution works perfectly, and I’ve successfully integrated it into my project.
Much appreciated!
